File: tests/0414_third_maximum_number_test.py
import importlib
import pytest

module = importlib.import_module("algorithms.0414_third_maximum_number")


def test_third_max():
    s = module.Solution()
    assert s.thirdMax([3, 2, 1]) == 1
    assert s.thirdMax([1, 2]) == 2
    assert s.thirdMax([2, 2, 3, 1]) == 1

File: tests/0414_third_maximum_number_test.cpp
#include "0414_third_maximum_number.cpp"
#include <doctest.h>

TEST_CASE("thirdMax")
{
    Solution s = Solution();
    std::vector<int> nums{3, 2, 1};
    CHECK(s.thirdMax(nums) == 1);

    nums = {1, 2};
    CHECK(s.thirdMax(nums) == 2);

    nums = {2, 2, 3, 1};
    CHECK(s.thirdMax(nums) == 1);
}
